src/VisualStudio/Core/Def/NavigateTo/RoslynSearchResultViewFactory.cs

```diff
@@ -64,9 +64,9 @@ public Task<IReadOnlyList<SearchResultPreviewPanelBase>> GetPreviewPanelsAsync(S
                 return null;
 
             Uri? absoluteUri;
-            if (document.IsSourceGeneratedDocument)
+            if (document.SourceGeneratedDocumentIdentity is not null)
```

Comment:
If we had to pass through the identity just so we can create a URI, why not just have the document object just hold onto the URI in the first place? Then we can delete this handling here in favor of a document.GetURI() when the object is created?

Reply:
the code creating the NavigableDocument is in features, which doesn't have any of the URI stuff from the LSP layer